c11threads.h

```diff
@@ -60,25 +123,164 @@ enum {
+static inline int cnd_signal(cnd_t *cond);
+static inline int cnd_broadcast(cnd_t *cond);
+static inline int cnd_wait(cnd_t *cond, mtx_t *mtx);
+static inline int cnd_timedwait(cnd_t *cond, mtx_t *mtx, const struct timespec *ts);
+#endif
+
+/* Thread-specific storage functions. */
+
+static inline int tss_create(tss_t *key, tss_dtor_t dtor);
+static inline void tss_delete(tss_t key);
+static inline int tss_set(tss_t key, void *val);
+static inline void *tss_get(tss_t key);
+
+#if !defined(_WIN32) || defined(C11THREADS_PTHREAD_WIN32) || !defined(C11THREADS_SUPPORT_WINNT_OLDER_THAN_VISTA)
+/* One-time callable function. */
+
+static inline void call_once(once_flag *flag, void (*func)(void));
+#endif
+
+#ifdef C11THREADS_NO_TIMESPEC_GET
+#ifndef UTC_TIME
+#define UTC_TIME 1
+#endif
+static inline int timespec_get(struct timespec *ts, int base);
+#endif
+
+
+/* ---- platform ---- */
+
+#if defined(_WIN32) && !defined(C11THREADS_PTHREAD_WIN32)
+void _thrd_init_win32(void);
+void _thrd_destroy_win32(void);
+
+int _thrd_self_register_win32(void);
+int _thrd_create_win32(thrd_t *thr, thrd_start_t func, void *arg);
+void _thrd_exit_win32(int res);
+int _thrd_join_win32(thrd_t thr, int *res);
+int _thrd_detach_win32(thrd_t thr);
+thrd_t _thrd_current_win32(void);
+int _thrd_sleep_win32(const struct timespec *ts_in, struct timespec *rem_out);
+void _thrd_yield_win32(void);
+
+int _mtx_init_win32(mtx_t *mtx, int type);
+void _mtx_destroy_win32(mtx_t *mtx);
+int _mtx_lock_win32(mtx_t *mtx);
+int _mtx_trylock_win32(mtx_t *mtx);
+int _mtx_timedlock_win32(mtx_t *mtx, const struct timespec *ts);
+int _mtx_unlock_win32(mtx_t *mtx);
+
+#if !defined(_WIN32) || defined(C11THREADS_PTHREAD_WIN32) || !defined(C11THREADS_SUPPORT_WINNT_OLDER_THAN_VISTA)
+int _cnd_init_win32(cnd_t *cond);
+int _cnd_signal_win32(cnd_t *cond);
+int _cnd_broadcast_win32(cnd_t *cond);
+int _cnd_wait_win32(cnd_t *cond, mtx_t *mtx);
+int _cnd_timedwait_win32(cnd_t *cond, mtx_t *mtx, const struct timespec *ts);
+#endif
+
+int _tss_create_win32(tss_t *key, tss_dtor_t dtor);
+void _tss_delete_win32(tss_t key);
+int _tss_set_win32(tss_t key, void *val);
+void *_tss_get_win32(tss_t key);
+
+#if !defined(_WIN32) || defined(C11THREADS_PTHREAD_WIN32) || !defined(C11THREADS_SUPPORT_WINNT_OLDER_THAN_VISTA)
+void _call_once_win32(once_flag *flag, void (*func)(void));
+#endif
+
+#ifdef C11THREADS_NO_TIMESPEC_GET
+int _timespec_get_win32(struct timespec *ts, int base);
+#endif
+#endif
+
+/* ---- globals ---- */
+
+static inline void thrd_init(void)
+{
+#if defined(_WIN32) && !defined(C11THREADS_PTHREAD_WIN32)
+	_thrd_init_win32();
+#endif
+}
+
+static inline void thrd_destroy(void)
+{
+#if defined(_WIN32) && !defined(C11THREADS_PTHREAD_WIN32)
+	_thrd_destroy_win32();
+#endif
+}
 
 /* ---- thread management ---- */
 
+static inline int thrd_self_register(void)
+{
+#if defined(_WIN32) && !defined(C11THREADS_PTHREAD_WIN32)
+	return _thrd_self_register_win32();
+#endif
```

Comment:
There's no return for the non-win32 version. But is this function necessary at all? And if it's necessary only for the win32 code, maybe it should be moved into c11trheads_win32.c

Reply:
Correct, I forgot to add the `return`. The function is in here to make it available by including the header. I could just have a declaration here, but felt it was cleaner to have a public function and an implementation.

Reply:
Those functions were moved. Solved.